test/Libraries/Microsoft.Extensions.TimeProvider.Testing.Tests/TimerTests.cs

```diff
@@ -315,7 +318,10 @@ public void MultipleTimersCallbackInvokedInScheduledOrder()
         using var timer2 = timeProvider.CreateTimer(_ => callbacks.Add((2, timeProvider.GetElapsedTime(startTime))), null, TimeSpan.FromMilliseconds(3), TimeSpan.FromMilliseconds(3));
         using var timer3 = timeProvider.CreateTimer(_ => callbacks.Add((3, timeProvider.GetElapsedTime(startTime))), null, TimeSpan.FromMilliseconds(6), TimeSpan.FromMilliseconds(5));
 
-        timeProvider.Advance(TimeSpan.FromMilliseconds(11));
+        timeProvider.Advance(TimeSpan.FromMilliseconds(3));
+        timeProvider.Advance(TimeSpan.FromMilliseconds(3));
+        timeProvider.Advance(TimeSpan.FromMilliseconds(3));
+        timeProvider.Advance(TimeSpan.FromMilliseconds(2));
```

Comment:
We want to move time forward by 11ms in one step. That should still make the assertion pass below. The fake time provider should call the timer callbacks in order, with the internal `_now` set correctly at each callback time, independent of whether we move past one or more timer "period" callbacks.

Reply:
Sorry, but this is fundamentally incorrect. There is no guarantee that the perceived wake time of a timer callback matches the specified due time. That's not how timers work.